### ISSUE TITLE
переименования аффиксов

### ISSUE THREAD

**Issue by veramal** [tuffnatty/suddenly#201]
Нужно переименовать некоторые аффиксы - в документе, в парсере и таблице поиска.
Общие принципы: 
1) подстрочные символы оставляем только для цифр, маркирующих одноименные слоты (Pl₁ ЛАр, Add₂ ТАА  и т.п.). В идеале обычные цифры используем только для обозначения лиц (1sg), но пока есть Dur1.
2) при указании на конкретный диалект слово dial убираем
Список будет пополняться, пока такой.
Abl₂ тЫн > Abl.arch
Abl₁ ДАң > Abl.dial
Acc₁ НЫ, Acc₂ Н - в парсере это все Acc без доп.знаков,  в таблице должно быть так же. Аналогично диалектные варианты Acc.dial₁ ТЫ, Acc.dial₂ Т
All.dial САрЫ | СА, All.dial₁ (н)САрЫ, All.dial₂ (н)СА - надо разбить на 2 кучи All.dial САрЫ | (н)САрЫ и All.dial СА | (н)СА. Можно их назвать их All.dial1 и All.dial2, но только не подстрочными цифрами.
Conv.pas.dial / Convпас А.бАс > Cv.kac Абас (без подстрочных и точки)
Convₚ (Ы)п > Cv.п
Conv.a А > Cv.a
Pres2.dial.kac чадыр | чады > Pres2.kac
Pres.dial.kyz тур | ту > Pres.kyz 
Pres.dial.sh чар > Pres.sh
Dur₁.dial.kac Ат >  Dur1.kac Ат (не подстрочная цифра)
Dur₁.dial.sag ит > Dur₁.dial.sag ит (не подстрочная цифра)
Dur1ᵢ и, Dur1ᵢᵣ ир > в идеале  надо бы в парсере объединить в Dur1, в поиске оставить возможность искать по отдельности. Но т.к. объединены Iter АдЫ | АдЫр, Pres2 чАдЫр | чАдЫ, Fut Ар | А и др., пусть и Dur1 будет такой же.

**Comment by tuffnatty** [comment]
> 1. подстрочные символы оставляем только для цифр, маркирующих одноименные слоты (Pl₁ ЛАр, Add₂ ТАА  и т.п.). В идеале обычные цифры используем только для обозначения лиц (1sg), но пока есть Dur1.

Соответственно, NF₀ -> NF.0 ?

>    Abl₁ ДАң > Abl.dial

То есть обычного Abl не остается, только диалектный?

>    Convₚ (Ы)п > Cv.п
>    Conv.a А > Cv.a

Мне не очень нравится использование кириллицы в тегах. Технически работать пока что будет, но отчего-то не нравится.

>    Dur₁.dial.sag ит > Dur₁.dial.sag ит (не подстрочная цифра)

Наверное, Dur1.sag , без .dial?

**Comment by tuffnatty** [comment]
Переименовал пока как есть

**Comment by veramal** [comment]
С аблативом ошиблась - есть обычный ДАң, а есть диалектный нАң.
Dur1.sag без .dial, да.
Конвербы - ну пусть будет латиницей, Cv.p.
NF₀ -> NF0 тогда уж, без точки. Но лучше просто NF - объединить с вариантом Ып под одним тегом, как _и_ и _ир_, допустим. Опять же, в поиске хорошо бы оставить 2 варианта, а в самом парсере один - если так можно.

**Comment by adybo** [comment]
Вера, хорошо бы точка в условных обозначениях (а зачем она?) отличалась от точки, обозначающей кумуляцию

**Comment by veramal** [comment]
Да, это интересный вопрос, надо подумать. Точка - чтобы различать два
слова. Не всякий почленит, если Pressh написать без точки.

**Comment by tuffnatty** [comment]
> С аблативом ошиблась - есть обычный ДАң, а есть диалектный нАң.

Стало совсем непонятно. Есть обычный ДАң, поссессивный нАң, диалектный обычный нАң и диалектный посессивный нАң?

> NF₀ -> NF0 тогда уж, без точки. Но лучше просто NF - объединить с вариантом Ып под одним тегом, как _и_ и _ир_, допустим.

Это можно.

> Опять же, в поиске хорошо бы оставить 2 варианта, а в самом парсере один - если так можно. 

Это нельзя.

Как лучше?

**Comment by veramal** [comment]
пн, 3 мая 2021 г. в 18:29, Phil Krylov ***@***.***>:

> С аблативом ошиблась - есть обычный ДАң, а есть диалектный нАң.
>
> Стало совсем непонятно. Есть обычный ДАң, поссессивный нАң, диалектный
> обычный нАң и диалектный посессивный нАң?
>
Нет, посессивный один.

> NF₀ -> NF0 тогда уж, без точки. Но лучше просто NF - объединить с
> вариантом Ып под одним тегом, как *и* и *ир*, допустим.
>
> Это можно.
>
> Опять же, в поиске хорошо бы оставить 2 варианта, а в самом парсере один -
> если так можно.
>
> Это нельзя.
>
> Как лучше?
>
Лучше, конечно, как можно - как нельзя все равно нельзя.

**Comment by tuffnatty** [comment]
> Нет, посессивный один.

ок

> Лучше, конечно, как можно - как нельзя все равно нельзя.

Можно - оставить отдельно NF0 и NF и в парсере, и в поиске, либо объединить их и в парсере, и в поиске.

**Comment by veramal** [comment]
Ну давайте попробуем объединить.
В таблице аффиксов вижу странное:

   -  All.dial СА | САрЫ
   -  All.dial1 (н)САрЫ
   -  All.dial2 (н)СА
   -


Надо или сделать один All.dial, или, что лучше, All.dial1 СА | (н) СА +
All.dial2 САрЫ | (н)САрЫ.

**Comment by veramal** [comment]
Да, еще в позиции 13 не Pl₂ ЛАр, а просто  Pl.

**Comment by veramal** [comment]
И почему-то Perf0 уехал в слот 5, хотя он должен быть в одном слоте с
обычным Perf.

**Comment by veramal** [comment]
Еще заметила: в таблице поиска нумерация слотов начинается с 0, а должна с
1 (т.к. 0 - это основа).
И хорошо бы уменьшить количество личных окончаний: т.к. варианты mix
совпадают либо с br, либо с full, лучше их убрать, а выбор варианта описать
правилами (mix = 1sg.br + др.лица в полных формах). В реальности во многих
формах первое лицо варьирует, поэтому получается лишняя омонимия (одно и то
же слово типа парғазың получает 2 разбора, в одном из которых 2sg.mix, в
другом - 2sg(.full)). Могу убрать из таблицы формы mix, если все согласны,
что можно без них.

**Comment by adybo** [comment]
Да, я согласна

**Comment by veramal** [comment]
Так, сейчас все вышесказанное сделано, кроме окончаний серии mix.
Из неудобных цифр остается проблема Perf0, Dur1 и Pres2. Возможно, стоит дать им всем один номер, поскольку в обоих случаях это вторая форма (вдобавок к дефолтной).
Есть также мысль точку, использующуюся некумулятивно, а следующее слово писать с большой буквы. Однако и сейчас различие есть: после кумулятивной точки слово пишется с большой буквы (Neg.Fut), а после некумулятивной - с маленькой (Distr.dial). Может, так и оставить?

**Comment by veramal** [comment]
Решили выбрать для дополнительных форм цифру 1, некумулятивную точку убрать. Убрать цифру 3 из глосс частиц в конце словоформы, потому что это их дефолтное место (аналогично Pl вместо Pl2). И еще некоторые мелочи. Таким образом, надо переназвать следующее:
Perf0 -> Perf1
Pres2 -> Pres1
NF.Neg -> Neg.NF (это все же не склейка с отрицанием ПА, которое перед временем, а отдельная позиция для отрицания)
NF.Neg.sh -> Neg.NFSh 
Dur1.sag -> Dur1Sag
Dur1.kac -> Dur1Kac
Dur.dial -> DurDial
Prosp.dial -> ProspDial
Distr.dial -> DistrDial
Pres.dial -> PresDial
PresPt.dial -> PresPtDial
Pres2.kac -> Pres1Kac
Pres.kyz -> PresKyz
Pres.sh -> PresSh
Cv.a -> CvA
Cv.p -> CvP
Cv.pas -> CvKac (потому что диалектная форма должна отсылать к диалекту, ср. Neg.NFSh)
Acc.dial -> AccDial
Dat.dial -> DatDial
Gen.dial -> GenDial
Instr.dial -> InstrDial
All.dial1 -> AllDial1
All.dial2 -> AllDial2
Delib.dial -> DelibDial
1pl.dial -> 1plDial
2pl.dial -> 2plDial
1sg.dial -> 1sgDial
Imp.1pl.dial -> Imp.1plDial
(про прочие императивные окончания еще думаем)
Ass₃ - > Ass

**Comment by veramal** [comment]
И еще Abl.arch -> AblArch

**Comment by veramal** [comment]
> в таблице поиска нумерация слотов начинается с 0, а должна с 1 (т.к. 0 - это основа).

Это сейчас так и надо исправить.


> И хорошо бы уменьшить количество личных окончаний: т.к. варианты mix совпадают либо с br, либо с full, лучше их убрать, а выбор варианта описать правилами (mix = 1sg.br + др.лица в полных формах). В реальности во многих формах первое лицо варьирует, поэтому получается лишняя омонимия (одно и то же слово типа парғазың получает 2 разбора, в одном из которых 2sg.mix, в другом - 2sg(.full)). Могу убрать из таблицы формы mix, если все согласны, что можно без них.

Это тоже уже пора. Зачеркнула в таблице окончания .mix.

> (про прочие императивные окончания еще думаем)

Нашла в письмах такое предложение: Imp.1.Incl заменить на ImpIncl, Imp.1pl.Incl заменить на ImpIncl.Pl. Исправила в таблице.

Еще в некоторый момент переименовали Affirm в Vis (визибилитив), поправила название позиции 18 в таблице.

**Comment by veramal** [comment]
Вернула вид (Ы)м (вместо м) показателю 1sg.br, т.к. если убрать серию mix, формы типа парарым 'уйду' анализироваться не будут.
Зачеркнула также 1sg.br ПЫс, поскольку он совпадает с 1sg ПЫс, а мы теперь хотим помечать краткую серию окончаний только в тех случаях, когда показатели отличаются от полной серии.

**Comment by tuffnatty** [comment]
> Зачеркнула также 1sg.br ПЫс, поскольку он совпадает с 1sg ПЫс, а мы теперь хотим помечать краткую серию окончаний только в тех случаях, когда показатели отличаются от полной серии.

не sg, а pl?

**Comment by veramal** [comment]
Да, pl конечно.

**Comment by tuffnatty** [comment]
Нуждается в обновлении ограничение 20, перестали разбираться:
адибыс < ада+а+быс (Fut А+1pl)
азырихчатхам < азыра+ах+чат+ха+м (Past+1sg)
килчам < кил+0̸+ча+м  (PresDial ча+1sg)
килчадым < кил+0̸+чады+м (Pres1Kac+1sg)
нанадырым < нан+адыр+ым (Gener(полн.)+1sg)
парам < пар+а+м (Fut А+1sg)
и др.

**Comment by veramal** [comment]
В правиле 20 все верно написано: "Cмешанный набор форм (1sg.br + др.лица в полных формах) присоединяется к аффиксам: Pres чА ..." и т.д. Надо научить парсер склеивать парадигму из указанных форм.

**Comment by tuffnatty** [comment]
Не стал учить, просто переименовал 1sg.mix в 1sg.br, а остальные .mix - без .mix, а правила оставил старые.

**Comment by veramal** [comment]
По старым правилам часто выходило два разбора: через full (которые непомечены) и через mix. Если сейчас не будет по 2 одинаковых разбора выдаваться, то ладно.

**Comment by adybo** [comment]
не надо ли тогда претерит лишить I?

**Comment by veramal** [comment]
В текущем описании нет ни претерита, ни I. Поясни конкретнее, что имеется в
виду?

> не надо ли тогда претерит лишить I?

**Comment by adybo** [comment]
прошедшее на -ды

  
>Пятница, 27 мая 2022, 0:05 +03:00 от veramal ***@***.***>:
> 
>  В текущем описании нет ни претерита, ни I. Поясни конкретнее, что имеется в
>виду?

**Comment by tuffnatty** [comment]
Можно закрывать?

**Comment by veramal** [comment]
Нет, т.к. проблема двух разборов не исччезла, а усугубилась - теперь они
совершенно одинаковые:
килчебіс <https://khakas.altaica.ru/grammar/#>
*v* килерге ‛приходить’ кил
--0̸------чА----------ПЫс---
--NF------Pres----------1pl---
*v* килерге ‛приходить’ кил
--0̸------чА----------ПЫс---
--NF------Pres----------1pl---

**Comment by tuffnatty** [comment]
Поправил, дубли убрались

**Comment by veramal** [comment]
Хорошо.

**Comment by veramal** [comment]
Убилось лишнее по дороге: 1pl.br ПЫс был ошибочно зачеркнут, в результате сейчас не разбираются полдыбыс и ползабыс. В документе исправила. Поправь в корпусе, пожалуйста.

**Comment by veramal** [comment]
И чего его предлагается лишить?

пт, 27 мая 2022 г. в 00:06, adybo ***@***.***>:

**Comment by veramal** [comment]
Проблема с окончаниями 1pl перенесена в топик https://github.com/tuffnatty/suddenly/issues/224. Отслаьное вроде решилось
